cli/azd/cmd/auth_login.go

```diff
@@ -143,6 +149,7 @@ type loginAction struct {
 	authManager       *auth.Manager
 	accountSubManager *account.SubscriptionsManager
 	flags             *loginFlags
+	annotations       CmdAnnotations
 }
 
 func newLoginAction(
```

Comment:
@wbreza @ellismg Still debugging further, but I think the underlying problem is that commands are no longer isolated from one another, when they share a single root `container`.

The flags I attached to `azd auth login` that was parsed by `cobra`, ended up going to the invisible command instance for `azd login`. At no point a runtime/compile panic occurred. The behavior here is wild, and not something I'd expect. I'm trying to come up with ways to solve this.

Reply:
Figured out that registration is unique to a particular type, so I'll introduce a new action and flag type to solve this for now. Adding tests for coverage since this seems important for VSCode.

Reply:
@bwateratmsft This is addressed in the latest iteration, with test coverage added for the scenario. Let me know if you find out differently.

Reply:
I verified that it's working correctly now. Thanks @weikanglim!